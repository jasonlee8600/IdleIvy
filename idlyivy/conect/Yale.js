import { Contract } from '@ethersproject/contracts';
import Yale from '../conect/Yale.json' 

export const contractAddress = '0xC57C5aC5cDBFe5d77A4Dd539205Bc07dF0930533';

export const getContract = (library, account) => {
	const signer = library.getSigner(account).connectUnchecked();
	var contract = new Contract(contractAddress, Yale.abi, signer);
	return contract;
};

 