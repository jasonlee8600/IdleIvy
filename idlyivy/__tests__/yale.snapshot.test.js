import Yale from '../pages/yale';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(<Yale />).toJSON();
    expect(tree).toMatchSnapshot();
  });