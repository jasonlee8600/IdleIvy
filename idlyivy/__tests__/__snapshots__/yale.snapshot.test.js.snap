// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="flex flex-row justify-between"
  >
    <div
      className="flex flex-col items-center hidden h-screen md:flex w-[250px] md:min-w-[250px] lg:min-w-[250px] fixed top-0 gap-2 shadow-2xl bg-gradient-to-b from-[#272727] to-[#797979] outline outline-white outline-[5px]"
    >
      <div
        className="fixed top-[2%]"
      >
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <h2
            className="text-white font-semibold text-[60px]"
          >
            IdleIvy
          </h2>
        </a>
      </div>
      <div
        className="fixed top-[15%]"
      >
        <img
          className="mt=[20px]"
          src="../yalelogo.svg"
        />
      </div>
      <div
        className="fixed top-[35%]"
      >
        <div
          className="flex flex-col items-center mt-[20px]"
        >
          <h2
            className="text-4xl text-white"
          >
            Tokens:
          </h2>
          <h3
            className="text-2xl text-white"
          >
            1,000,000
          </h3>
        </div>
        <div
          className="flex flex-col items-center mt-[50px]"
        >
          <h2
            className="text-4xl text-white"
          >
            Rate:
          </h2>
          <h3
            className="text-2xl text-white"
          >
            1 tokens/min
          </h3>
        </div>
        <button
          className="bg-[#202B64] mt-[50px] px-[35px] py-[10px] rounded-full outline outline-white outline-[7px]"
          id="collect"
          type="button"
        >
          <div
            className="flex flex-col items-center"
          >
            <h2
              className="text-4xl text-white"
            >
              Collect:
            </h2>
            <h3
              className="text-2xl text-white"
            >
              1,000
            </h3>
          </div>
        </button>
      </div>
      <div
        className="fixed top-[80%]"
      >
        <div
          className="flex flex-col items-center"
        >
          <h2
            className="text-white font-semibold text-[30px]"
          >
            Leaderboard
          </h2>
        </div>
        <div
          className="flex flex-col items-center mt-[20px]"
        >
          <a
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <h2
              className="text-white font-semibold text-[30px]"
            >
              Home
            </h2>
          </a>
        </div>
      </div>
    </div>
    <div
      className="flex flex-col w-full h-fit bg-[url('../public/yalebg.jpeg')] bg-cover md:ml-[250px] gap-[75px] pt-16"
    >
      <div
        className="flex flex-row ml-[25px]"
      >
        <img
          alt="Buttery"
          className="rounded-full outline outline-white outline-[4px]"
          data-nimg="1"
          decoding="async"
          height={NaN}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src=""
          style={
            {
              "color": "transparent",
            }
          }
          width={65}
        />
        <div
          className="flex flex-row w-full pl-4"
        >
          <div
            className="flex flex-row z-0 bg-gradient-to-b from-[#272727] to-[#797979] h-[65px] w-5/6 rounded-full outline outline-white outline-[4px]"
          >
            <button
              className="self-start z-10 bg-[#202B64] h-[65px] w-1/4 rounded-full outline outline-white outline-[4px] text-xl text-white"
              id="upgrade_buttery"
              type="button"
            >
              Upgrade
            </button>
            <div
              className="flex flex-col justify-center w-4/5"
            >
              <h3
                className="text-2xl text-white text-center"
              >
                Buttery
              </h3>
              <h4
                className="text-lg text-white text-center"
              >
                Rate: 1 token / minute
              </h4>
            </div>
          </div>
        </div>
      </div>
      <div
        className="flex flex-row ml-[25px]"
      >
        <img
          alt="Buttery"
          className="rounded-full outline outline-white outline-[4px]"
          data-nimg="1"
          decoding="async"
          height={NaN}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src=""
          style={
            {
              "color": "transparent",
            }
          }
          width={65}
        />
        <div
          className="flex flex-row w-full pl-4"
        >
          <div
            className="flex flex-row z-0 bg-gradient-to-b from-[#272727] to-[#797979] h-[65px] w-5/6 rounded-full outline outline-white outline-[4px]"
          >
            <button
              className="self-start z-10 bg-[#202B64] h-[65px] w-1/4 rounded-full outline outline-white outline-[4px] text-xl text-white"
              id="upgrade_buttery"
              type="button"
            >
              Upgrade
            </button>
            <div
              className="flex flex-col justify-center w-4/5"
            >
              <h3
                className="text-2xl text-white text-center"
              >
                Sterling Library
              </h3>
              <h4
                className="text-lg text-white text-center"
              >
                Rate: 10 tokens / minute
              </h4>
            </div>
          </div>
        </div>
      </div>
      <div
        className="flex flex-row ml-[25px]"
      >
        <img
          alt="Buttery"
          className="rounded-full outline outline-white outline-[4px]"
          data-nimg="1"
          decoding="async"
          height={NaN}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src=""
          style={
            {
              "color": "transparent",
            }
          }
          width={65}
        />
        <div
          className="flex flex-row w-full pl-4"
        >
          <div
            className="flex flex-row z-0 bg-gradient-to-b from-[#272727] to-[#797979] h-[65px] w-5/6 rounded-full outline outline-white outline-[4px]"
          >
            <button
              className="self-start z-10 bg-[#202B64] h-[65px] w-1/4 rounded-full outline outline-white outline-[4px] text-xl text-white"
              id="upgrade_buttery"
              type="button"
            >
              Upgrade
            </button>
            <div
              className="flex flex-col justify-center w-4/5"
            >
              <h3
                className="text-2xl text-white text-center"
              >
                Yale Bowl
              </h3>
              <h4
                className="text-lg text-white text-center"
              >
                Rate: 100 tokens / minute
              </h4>
            </div>
          </div>
        </div>
      </div>
      <div
        className="flex flex-row ml-[25px]"
      >
        <img
          alt="Buttery"
          className="rounded-full outline outline-white outline-[4px]"
          data-nimg="1"
          decoding="async"
          height={NaN}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src=""
          style={
            {
              "color": "transparent",
            }
          }
          width={65}
        />
        <div
          className="flex flex-row w-full pl-4"
        >
          <div
            className="flex flex-row z-0 bg-gradient-to-b from-[#272727] to-[#797979] h-[65px] w-5/6 rounded-full outline outline-white outline-[4px]"
          >
            <button
              className="self-start z-10 bg-[#202B64] h-[65px] w-1/4 rounded-full outline outline-white outline-[4px] text-xl text-white"
              id="upgrade_buttery"
              type="button"
            >
              Upgrade
            </button>
            <div
              className="flex flex-col justify-center w-4/5"
            >
              <h3
                className="text-2xl text-white text-center"
              >
                Handsome Dan
              </h3>
              <h4
                className="text-lg text-white text-center"
              >
                Rate: 1,000 tokens / minute
              </h4>
            </div>
          </div>
        </div>
      </div>
      <div
        className="flex flex-row ml-[25px]"
      >
        <img
          alt="Buttery"
          className="rounded-full outline outline-white outline-[4px]"
          data-nimg="1"
          decoding="async"
          height={NaN}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src=""
          style={
            {
              "color": "transparent",
            }
          }
          width={65}
        />
        <div
          className="flex flex-row w-full pl-4"
        >
          <div
            className="flex flex-row z-0 bg-gradient-to-b from-[#272727] to-[#797979] h-[65px] w-5/6 rounded-full outline outline-white outline-[4px]"
          >
            <button
              className="self-start z-10 bg-[#202B64] h-[65px] w-1/4 rounded-full outline outline-white outline-[4px] text-xl text-white"
              id="upgrade_buttery"
              type="button"
            >
              Upgrade
            </button>
            <div
              className="flex flex-col justify-center w-4/5"
            >
              <h3
                className="text-2xl text-white text-center"
              >
                Buttery
              </h3>
              <h4
                className="text-lg text-white text-center"
              >
                Rate: 1 token / minute
              </h4>
            </div>
          </div>
        </div>
      </div>
      <div
        className="flex flex-row ml-[25px]"
      >
        <img
          alt="Buttery"
          className="rounded-full outline outline-white outline-[4px]"
          data-nimg="1"
          decoding="async"
          height={NaN}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src=""
          style={
            {
              "color": "transparent",
            }
          }
          width={65}
        />
        <div
          className="flex flex-row w-full pl-4"
        >
          <div
            className="flex flex-row z-0 bg-gradient-to-b from-[#272727] to-[#797979] h-[65px] w-5/6 rounded-full outline outline-white outline-[4px]"
          >
            <button
              className="self-start z-10 bg-[#202B64] h-[65px] w-1/4 rounded-full outline outline-white outline-[4px] text-xl text-white"
              id="upgrade_buttery"
              type="button"
            >
              Upgrade
            </button>
            <div
              className="flex flex-col justify-center w-4/5"
            >
              <h3
                className="text-2xl text-white text-center"
              >
                Sterling Library
              </h3>
              <h4
                className="text-lg text-white text-center"
              >
                Rate: 10 tokens / minute
              </h4>
            </div>
          </div>
        </div>
      </div>
      <div
        className="flex flex-row ml-[25px]"
      >
        <img
          alt="Buttery"
          className="rounded-full outline outline-white outline-[4px]"
          data-nimg="1"
          decoding="async"
          height={NaN}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src=""
          style={
            {
              "color": "transparent",
            }
          }
          width={65}
        />
        <div
          className="flex flex-row w-full pl-4"
        >
          <div
            className="flex flex-row z-0 bg-gradient-to-b from-[#272727] to-[#797979] h-[65px] w-5/6 rounded-full outline outline-white outline-[4px]"
          >
            <button
              className="self-start z-10 bg-[#202B64] h-[65px] w-1/4 rounded-full outline outline-white outline-[4px] text-xl text-white"
              id="upgrade_buttery"
              type="button"
            >
              Upgrade
            </button>
            <div
              className="flex flex-col justify-center w-4/5"
            >
              <h3
                className="text-2xl text-white text-center"
              >
                Yale Bowl
              </h3>
              <h4
                className="text-lg text-white text-center"
              >
                Rate: 100 tokens / minute
              </h4>
            </div>
          </div>
        </div>
      </div>
      <div
        className="flex flex-row ml-[25px]"
      >
        <img
          alt="Buttery"
          className="rounded-full outline outline-white outline-[4px]"
          data-nimg="1"
          decoding="async"
          height={NaN}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src=""
          style={
            {
              "color": "transparent",
            }
          }
          width={65}
        />
        <div
          className="flex flex-row w-full pl-4"
        >
          <div
            className="flex flex-row z-0 bg-gradient-to-b from-[#272727] to-[#797979] h-[65px] w-5/6 rounded-full outline outline-white outline-[4px]"
          >
            <button
              className="self-start z-10 bg-[#202B64] h-[65px] w-1/4 rounded-full outline outline-white outline-[4px] text-xl text-white"
              id="upgrade_buttery"
              type="button"
            >
              Upgrade
            </button>
            <div
              className="flex flex-col justify-center w-4/5"
            >
              <h3
                className="text-2xl text-white text-center"
              >
                Handsome Dan
              </h3>
              <h4
                className="text-lg text-white text-center"
              >
                Rate: 1,000 tokens / minute
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
